simapp/abci.go

```diff
@@ -2,69 +2,179 @@ package simapp
 
 import (
 	"bytes"
-	"crypto/rand"
 	"encoding/json"
 	"fmt"
 
 	abci "github.com/cometbft/cometbft/abci/types"
 
 	"github.com/cosmos/cosmos-sdk/baseapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 )
 
+/*
+1. PrepareProposal -> Verify Prev. Block's VEs (check for 2/3+ majority ) -> Encode VEs in tx bytes
+2. ProcessProposal -> WE CAN execute txs -> resp. tx handler will set data buffer (on which we will eventually vote upon). (ALTERNATIVELY maybe we can do this in FinalizeBlock ?)
+3. ExtendVote -> Fetch data from buffer -> Vote on it acc. -> Store result.
+4. VerifyVoteExtension -> Verify Vote recieved from other validators and ACCEPT/REJECT acc.
+5. PreFinalizeBlock -> Extract VEs from prev. block and persist in DB.
+6. FinalizeBlock -> Finalize and execute txs.
+7. CommitBlock -> Self explanatory.
+*/
+
```

Comment:
@Raneet10 is this the confirmed flow?
@VAIBHAVJINDAL3012 can you add your concerns here wrt "sepcial tx"?

Reply:
Yes, Encoding the VE and putting it in a block is a special tx. I meant the same.